lib/trino-hive-formats/src/test/java/io/trino/hive/formats/ion/TestIonFormat.java

```diff
@@ -260,6 +300,216 @@ private void assertValues(RowType rowType, String ionText, List<Object>... expec
+            ObjectInspector fieldObjectInspector = field.getFieldObjectInspector();
+            switch (column.type()) {
+                case VarcharType _ -> {
+                    String result =
+                            ((IonValueToStringObjectInspector) fieldObjectInspector).getPrimitiveJavaObject(fieldValue);
+                    Assertions.assertEquals(result, expected);
+                    return;
+                }
+                case DecimalType _ -> {
+                    if (fieldValue instanceof HiveDecimal decimal) {
+                        DecimalType decimalType = (DecimalType) type;
+                        // writable messes with the scale so rescale the values to the Trino type
+                        BigDecimal bigDecimal = decimal.bigDecimalValue();
+                        bigDecimal = bigDecimal.setScale(decimalType.getScale(), HALF_UP);
+                        if (bigDecimal.precision() > decimalType.getPrecision()) {
+                            throw new IllegalArgumentException("decimal precision larger than column precision");
+                        }
+                        Assertions.assertEquals(bigDecimal, expected);
+                        return;
+                    } else if (fieldValue instanceof IonDecimal decimal) {
+                        HiveDecimal hiveDecimal =
+                                ((IonNumberToDecimalObjectInspector) fieldObjectInspector).getPrimitiveJavaObject(decimal);
+                        DecimalType decimalType = (DecimalType) type;
+                        // writable messes with the scale so rescale the values to the Trino type
+                        BigDecimal bigDecimal = hiveDecimal.bigDecimalValue();
+                        bigDecimal = bigDecimal.setScale(decimalType.getScale(), HALF_UP);
+                        if (bigDecimal.precision() > decimalType.getPrecision()) {
+                            throw new IllegalArgumentException("decimal precision larger than column precision");
+                        }
+                        Assertions.assertEquals(bigDecimal, expected);
+                        return;
+                    }
+                }
+                case TinyintType _ -> {
+                    Object result =
+                            ((IonIntToTinyIntObjectInspector) fieldObjectInspector).getPrimitiveJavaObject(fieldValue);
+                    Assertions.assertEquals(result, expected);
+                }
+                //TODO: Modify this to add more cases when adding more tests for a trino type
+                default -> throw new IllegalStateException("Unexpected value: " + column.type());
+            }
+
+        }
+    }
+
+    private static void assertValueFails(Type type, String ionValue) {
+        assertValueFailsHive(type, ionValue, false);
+        assertValueFailsTrino(type, ionValue);
+    }
+
+    private static void assertValueFails(Type type, String ionValue, boolean failOnOverflow)
```

Comment:
For the `assertValueFails`, we are expecting the error to be thrown for the mistyping or overflow cases. We might consider setting the value of `failOnOverflow` to `true` instead of making it configurable.